wire-library/wire-java-generator/src/test/java/com/squareup/wire/java/JavaGeneratorTest.java

```diff
@@ -563,7 +563,7 @@ public void generateTypeUsesPackageNameOnFieldAndClassNameClashWithinPackage() t
       .build();
     assertThat(new JavaWithProfilesGenerator(schema)
       .generateJava("common.proto.A"))
-        .contains("public final AnotherStatus common_proto_Status;");
+        .contains("public final AnotherStatus Status;");
```

Comment:
I think this is from the _second_ check in the `if` condition linked in previous comment. Now that we have the correct fully-qualified name of the field (`common.proto.A.Status`), it no longer conflicts with the name of the enum (`common.proto.Status`).

Like above, I don't understand _why_ we were doing this, so I'd like guidance on whether this change is a "fix" or if we need to change the generator to preserve the previous behavior.

Reply:
I think that's a regression. The test case is for this specific scenario. Here is the original PR: https://github.com/square/wire/pull/1318

Maybe we could create a golden file of this and make sure it compiles.

Reply:
Hmmm. That PR has no description 😢 

If this is a compile error, then the check that was originally here was wrong. The proto language already prevents a field from having the same name as a nested type, so there should not be an issue there. (In this case, the `Status` type is defined at a different lexical scope in the file.)

Perhaps the issue is just when generating Java code, types are unconditionally referred to without qualifiers? In the example here, inside the scope of class `A`, references to the enum would need to use the qualified name `common.proto.Status`, to disambiguate from the field named `Status`. (Seems weird to be renaming the field instead.)

Reply:
Hmm, I'm not really sure. I can add a golden file from the test once this PR merges and I'll ping you if this doesn't compile. Or we can add the golden file right now and add it somewhere in `gen-tests.gradle.kts`.

Reply:
How about I change this so the old behavior is still used, but add a TODO that it smells incomplete -- the logic that is there may have solved the issue at hand, but I strongly suspect there are still edge cases that could cause compilation issues. And I also think there are alternative ways around it that are more user-friendly than mangling their field names (IMO, that should only be necessary when field names conflict with language keywords).

Reply:
I reverted these assertions in the tests and updated the generators to preserve prior behavior here.

Reply:
👍